core/store/src/trie/mod.rs

```diff
@@ -800,6 +890,58 @@ mod tests {
     type TrieChanges = Vec<(Vec<u8>, Option<Vec<u8>>)>;
     const SHARD_VERSION: u32 = 1;
 
+    impl Trie {
+        pub fn verify_proof(
+            &self,
+            levels: Vec<TrieProofItem>,
+            value: &[u8],
+            expected_hash: CryptoHash,
```

Comment:
shouldn't you also pass the key here?

(as otherwise, from what I see, you only verify the existence of a given VALUE, but not the key that it belongs to).

Reply:
+1 to question. We also need a test for this - checking that verification fails if proof proves correct value but uses incorrect key.